configure.ac

```diff
@@ -584,68 +610,54 @@ case $libdir in
         ;;
 esac
 
-dnl Expand autoconf variables so that we don't end up with '${prefix}'
-dnl in #defines and python scripts
-dnl NOTE: Autoconf deliberately leaves them unexpanded to allow
-dnl    make exec_prefix=/foo install
-dnl No longer being able to do this seems like no great loss to me...
-
-eval prefix="`eval echo ${prefix}`"
-eval exec_prefix="`eval echo ${exec_prefix}`"
-eval bindir="`eval echo ${bindir}`"
-eval sbindir="`eval echo ${sbindir}`"
-eval libexecdir="`eval echo ${libexecdir}`"
-eval datadir="`eval echo ${datadir}`"
-eval sysconfdir="`eval echo ${sysconfdir}`"
-eval sharedstatedir="`eval echo ${sharedstatedir}`"
-eval localstatedir="`eval echo ${localstatedir}`"
-eval libdir="`eval echo ${libdir}`"
-eval includedir="`eval echo ${includedir}`"
-eval oldincludedir="`eval echo ${oldincludedir}`"
-eval infodir="`eval echo ${infodir}`"
-eval mandir="`eval echo ${mandir}`"
+dnl Expand values of autoconf-provided directory options
+expand_path_option prefix
+expand_path_option exec_prefix
+expand_path_option bindir
+expand_path_option sbindir
+expand_path_option libexecdir
+expand_path_option datadir
+expand_path_option sysconfdir
+expand_path_option sharedstatedir
+expand_path_option localstatedir
+expand_path_option libdir
+expand_path_option includedir
+expand_path_option oldincludedir
+expand_path_option infodir
+expand_path_option mandir
 
 dnl Home-grown variables
 
-if [ test "x${runstatedir}" = "x" ]; then
-    if [ test "x${pcmk_runstatedir}" = "x" ]; then
-        runstatedir="${localstatedir}/run"
-    else
-        runstatedir="${pcmk_runstatedir}"
-    fi
-fi
-eval runstatedir="$(eval echo ${runstatedir})"
+AS_IF([test x"${runstatedir}" = x""], [runstatedir="${pcmk_runstatedir}"])
+expand_path_option runstatedir "${localstatedir}/run"
 AC_DEFINE_UNQUOTED([PCMK_RUN_DIR], ["$runstatedir"],
                    [Location for modifiable per-process data])
```

Comment:
one more reason why we should step away from identifying sbd-daemon by pid-file. everything is kind of fine as long as we have /run duplicated at /var/run. but then ...